src/main/java/org/primefaces/extensions/component/fuzzysort/FuzzysortRenderer.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2011-2019 PrimeFaces Extensions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.extensions.component.fuzzysort;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.faces.FacesException;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+import javax.faces.render.FacesRenderer;
+import org.primefaces.renderkit.CoreRenderer;
+import org.primefaces.util.WidgetBuilder;
+
+@FacesRenderer(componentFamily = Fuzzysort.COMPONENT_FAMILY, rendererType = Fuzzysort.DEFAULT_RENDERER)
+public class FuzzysortRenderer extends CoreRenderer {
+
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
+        if (context == null) {
+            throw new NullPointerException("No context defined!");
+        }
+
+        Fuzzysort fuzzysort = (Fuzzysort) component;
+
+        encodeMarkup(context, fuzzysort);
+        encodeScript(context, fuzzysort);
+    }
+
+    private void encodeScript(FacesContext context, Fuzzysort fuzzysort) throws IOException {
+        WidgetBuilder wb = getWidgetBuilder(context);
+
+        // TODO is it possible to make improvement here to have @FuzzysortKey annotated class?
+        List list = (List) fuzzysort.getValue();
+        Object o = list.get(0);
+        Class<? extends Object> clazz = o.getClass();
+        Field[] fields = clazz.getDeclaredFields();
+        List<String> keys = Arrays.asList(fields).stream()
+                .filter(f -> f.getAnnotation(FuzzysortKey.class) != null)
+                .map(f -> f.getName())
+                .collect(Collectors.toList());
+        if (keys.isEmpty()) {
+            throw new FacesException("No @FuzzyKey annotation was detected on your class " + clazz + ". Please annotate at least one field in the class as @FuzzysortKey.");
+        }
+
+        String jsonKeys = new Gson().toJson(keys, new TypeToken<List<String>>() {
```

Comment:
@melloware was Gson already part of PFE? If not, I think it's better to switch to Json-B, as it is part of EE8.

Reply:
.. it's already used by many components

Reply:
Hi @jepsar , is it what you mean?

```java
Jsonb jsonb = JsonbBuilder.create();
//String jsonKeys = new Gson().toJson(keys, new TypeToken<List<String>>() {}.getType());
String jsonKeys = jsonb.toJson(keys, List.class);
//String jsonValue = new Gson().toJson(fuzzysort.getValue(), new TypeToken<List<Object>>() {}.getType());
String jsonValue = jsonb.toJson(fuzzysort.getValue(), List.class);
```

Reply:
@aripddev Yeah, but never mind this comment, as Gson is already used you should be able to use it as well. I already marked it as resolved.